src/app/controller/new.ts

```diff
@@ -19,7 +19,7 @@ class CreateAppController extends Controller {
       throw new Forbidden('Only users with the "admin" privilege can create new users');
     }
     ctx.response.type = 'text/html';
-    ctx.response.body = createAppForm(ctx.query.msg, ctx.query.error);
+    ctx.response.body = createAppForm(ctx.query.msg, ctx.query.error, ctx.query);
```

Comment:
Can you pass the `name` and `url` queries separately like the `msg` and `error` ones? Then, we can type those function parameters.

Reply:
Will do!

Reply:
Done :)